File: src_tests/simple_generate.py
#!/usr/bin/env python

import random
import string
import sys

random.seed(42)

assert len(sys.argv) == 4

items = int(sys.argv[1])
key_mu = int(sys.argv[2])
value_mu = int(sys.argv[3])


my_gauss = lambda mu: max(1,int(random.gauss(mu, mu * 0.35)))
rand_str = lambda l: ''.join(random.choice(string.lowercase) \
                                 for i in xrange(my_gauss(l)))

actions = [
    'set', 'set', 'set', 'set', 'set', 'set', 'set', 'set', 'set', 'set',
    'set', 'set', 'set', 'set', 'set', 'set', 'set', 'set', 'set', 'set',
    'del', 'del', 'del', 'del',
    'write',
    ]

for i in xrange(items):
    action = random.choice(actions)
    if action is 'set':
        print "set", rand_str(key_mu), \
            '%x' % (random.randint(0, 1<<16),) + '_' * my_gauss(value_mu)
    elif action is 'del':
        print "del", rand_str(key_mu)
    elif action is 'write':
        print "write"
    else:
        assert 0

File: src_tests/simple_generate.c
#include <assert.h>
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

char *key;
char *value;

static char *_rand_str(char *dst, int sz)
{
	int i;
	for (i=0; i < sz; i++) {
		dst[i] = 'a' + random() % 26;
	}
	return dst;
}

char *gen_key(unsigned key_sz)
{
	return _rand_str(key, key_sz);
}

char *gen_value(unsigned value_sz)
{
	if (value_sz > 6) {
		memset(&value[6], 'X', value_sz - 6);
		value_sz = 6;
	}
	return _rand_str(value, value_sz);
}

int main(int argc, char **argv)
{
	unsigned key_sz = 4;
	if (argc > 1) {
		key_sz = atoi(argv[1]);
		assert(key_sz > 0);
	}
	unsigned value_sz = 4;
	if (argc > 1) {
		value_sz = atoi(argv[2]);
	}
	key = malloc(key_sz + 1);
	value = malloc(value_sz + 1);

	fprintf(stderr, "key_sz=%u value_sz=%u\n", key_sz, value_sz);
	while (1) {
		switch(random() % 25) {
		case 0:
			printf("write\n");
			break;
		case 1:
		case 2:
		case 3:
		case 4:
			printf("del %s\n", gen_key(key_sz));
			break;
		default:
			printf("set %s %s\n", gen_key(key_sz),
			       gen_value(value_sz));
			break;
		}
	}
	return 0;
}
